db/db_impl_write.cc

```diff
@@ -1226,17 +1246,53 @@ Status DBImpl::ThrottleLowPriWritesIfNeeded(const WriteOptions& write_options,
 }
 
 Status DBImpl::ScheduleFlushes(WriteContext* context) {
-  ColumnFamilyData* cfd;
-  while ((cfd = flush_scheduler_.TakeNextColumnFamily()) != nullptr) {
-    auto status = SwitchMemtable(cfd, context, FlushReason::kWriteBufferFull);
-    if (cfd->Unref()) {
-      delete cfd;
+  if (atomic_flush_) {
+    if (flush_scheduler_.Empty()) {
+      return Status::OK();
     }
-    if (!status.ok()) {
-      return status;
+    Status s;
+    GroupFlushRequest group_flush_req;
+    for (auto cfd : *versions_->GetColumnFamilySet()) {
```

Comment:
this iteration may also be unsafe due to SwitchMemtable releasing/acquiring the db lock

Reply:
See my response to a similar comment.